deepdiff/diff.py

```diff
@@ -709,7 +709,7 @@ def _diff_iterable_in_order(self, level, parents_ids=frozenset(), _original_type
                     x,
                     y,
                     child_relationship_class=child_relationship_class,
-                    child_relationship_param=i)
+                    child_relationship_param=j)
```

Comment:
This change ensures that all indexes in the resulting diff are relative the t2 which is consistent with items added. If you are ignoring order I don't think this changes anything since `i` should be equal to `j`. But if using a compare function and not ignoring order we want the output indexes to be relative to t2 to make them able to be replayed. Only items removed are relative to t1 (since they have to be) and removed items always get replayed first in the delta objects.